java/code/src/com/redhat/rhn/domain/server/ServerFactory.java

```diff
@@ -1491,4 +1493,19 @@ public static int setMaintenanceScheduleToSystems(MaintenanceSchedule schedule,
                 .setParameter("systemIds", systemIds)
                 .executeUpdate();
     }
+
+    /**
+     * Remove MgrServerInfo from minion
+     *
+     * @param minion the minion
+     */
+    public static void dropMgrServerInfo(MinionServer minion) {
+        MgrServerInfo serverInfo = minion.getMgrServerInfo();
+        if (serverInfo != null) {
```

Comment:
You can invert this condition and do an early out to avoid nesting.

Reply:
changed